secure.go

```diff
@@ -198,12 +242,13 @@ func (s *Secure) Process(w http.ResponseWriter, r *http.Request) error {
 		}
 
 		http.Redirect(w, r, url.String(), status)
-		return fmt.Errorf("Redirecting to HTTPS")
+		return fmt.Errorf("redirecting to HTTPS")
 	}
 
+	s.responseHeader = make(http.Header)
```

Comment:
This will result in a race condition. If 20 requests come in, all 20 will make a new `http.Header` and over write it for other requests. We'll need to move this variable to a local instance so that it's isolated per request. We can either pass in another parameter to tell the function to send the headers or not, or we can return the `http.Header` and let the `Process` function handle the writing.